### PR TITLE
Add SwiftUI performance quickstart

### PR DESCRIPTION
**Pull request by jrcrespoh**
TODO:
- [x] Fix CI
- [x] Remove macOS platform
- [x] Update platform versions
- [ ] Add upload functionality
- [ ] Add UI tests
- [ ] Add comments
- [ ] Consider adding Performance custom metrics
- [ ] Consider restructuring data flow
- [ ] Consider vision

### CODE REVIEW

File: performance/SwiftUIPerformanceQuickstart/PerformanceExample/Shared/Process.swift
Comment (morganchen12) [inline performance/SwiftUIPerformanceQuickstart/PerformanceExample/Shared/Process.swift:71]: nit: print the error if the http response isn't a 2xx code.

File: performance/SwiftUIPerformanceQuickstart/PerformanceExample/Shared/Process.swift
Comment (morganchen12) [inline performance/SwiftUIPerformanceQuickstart/PerformanceExample/Shared/Process.swift:57]: nit: put the else on the same line as the guard

File: performance/SwiftUIPerformanceQuickstart/PerformanceExample/Shared/Process.swift
Comment (morganchen12) [inline performance/SwiftUIPerformanceQuickstart/PerformanceExample/Shared/Process.swift:21]: Maybe name this `LongRunningProcess`? Also, do you think it'd be a good idea to separate the class responsible for uploading the image from the class responsible for classifying it?

File: performance/SwiftUIPerformanceQuickstart/PerformanceExample/Shared/Process.swift
Comment (morganchen12) [inline performance/SwiftUIPerformanceQuickstart/PerformanceExample/Shared/Process.swift:27]: This is only used in one function, so instead of creating it on initialization and retaining it you should create and use it in the function itself. This will prevent issues like accidentally reusing traces that have already been stopped (for example if you call `downloadImageAsync()` twice).

**Review by morganchen12 — COMMENTED** [review]
Nice work! I left some feedback, but please wait for Maksym/Paul's review before merging.

File: .github/workflows/performance.yml
Comment (jrcrespoh) [reply .github/workflows/performance.yml]: Done ✅

File: performance/SwiftUIPerformanceQuickstart/PerformanceExample/PerformanceExample.xcodeproj/project.pbxproj
Comment (jrcrespoh) [reply performance/SwiftUIPerformanceQuickstart/PerformanceExample/PerformanceExample.xcodeproj/project.pbxproj:24]: Since I will be squashing and merging, any intermediary changes in these commits will not show up in the diff (since this is a new directory all files committed will be wholly new).

File: performance/SwiftUIPerformanceQuickstart/PerformanceExample/Shared/MainView.swift
Comment (jrcrespoh) [reply performance/SwiftUIPerformanceQuickstart/PerformanceExample/Shared/MainView.swift:19]: Done ✅

File: performance/SwiftUIPerformanceQuickstart/PerformanceExample/Shared/MainView.swift
Comment (jrcrespoh) [reply performance/SwiftUIPerformanceQuickstart/PerformanceExample/Shared/MainView.swift:32]: My understanding is that this would make it necessary to use Xcode 13 even if compiling for iOS/tvOS 14.x, which is supported. Our CI wouldn't be able to compile the project either. I wanted to avoid the duplication with the else branch of the `#available` iOS/tvOS version check, but I can simply let the else branch of the `#if` swift version check be `process.classifyImage()` which I imagine would be less disruptive, so I'll take this approach unless you suggest otherwise.

Comment (morganchen12) [reply performance/SwiftUIPerformanceQuickstart/PerformanceExample/Shared/MainView.swift:32]: Nah, let's just leave it as-is. Thanks for clarifying!

Comment (jrcrespoh) [reply performance/SwiftUIPerformanceQuickstart/PerformanceExample/Shared/MainView.swift:32]: We would indeed have a problem with the current code if for some reason we are compiling for iOS/tvOS 15 but using Xcode 12 (which I don't think is possible?) or setting Xcode 13 to compile using lower versions of Swift (I thought it too unlikely), so I'll just add the extra else branch just in case.

File: performance/SwiftUIPerformanceQuickstart/PerformanceExample/Shared/Process.swift
Comment (jrcrespoh) [reply performance/SwiftUIPerformanceQuickstart/PerformanceExample/Shared/Process.swift:57]: Done ✅

File: performance/SwiftUIPerformanceQuickstart/PerformanceExample/Shared/Process.swift
Comment (jrcrespoh) [reply performance/SwiftUIPerformanceQuickstart/PerformanceExample/Shared/Process.swift:71]: I'm printing the response since I don't believe I have access to an error in that scope. My understanding is that I only have access to an error if the `try await` function call throws an error.

File: performance/SwiftUIPerformanceQuickstart/PerformanceExample/Shared/Process.swift
Comment (jrcrespoh) [reply performance/SwiftUIPerformanceQuickstart/PerformanceExample/Shared/Process.swift:21]: I didn't implement this because I felt the classification task was really just one "relatively simple" function, to which the `precision` user setting is passed. The `categories` is more associated with the image than with a `Classifier`. Additionally, this would require message passing between `Classifier` and `LongRunningProcess` to update the status of the ideally "sole running task". Doing this might make the app more complicated than it needs to be? I would appreciate hearing your thoughts on this, especially if you were thinking of some advantages to separating them.

File: performance/SwiftUIPerformanceQuickstart/PerformanceExample/Shared/Process.swift
Comment (jrcrespoh) [reply performance/SwiftUIPerformanceQuickstart/PerformanceExample/Shared/Process.swift:27]: Done ✅

File: performance/SwiftUIPerformanceQuickstart/PerformanceExample/PerformanceExample.xcodeproj/project.pbxproj
Comment (paulb777) [reply performance/SwiftUIPerformanceQuickstart/PerformanceExample/PerformanceExample.xcodeproj/project.pbxproj:24]: `pod deintegrate` still needs to be run to avoid adding Pods references to the repo.

File: performance/SwiftUIPerformanceQuickstart/PerformanceExample/Podfile
Comment (paulb777) [inline performance/SwiftUIPerformanceQuickstart/PerformanceExample/Podfile:4]: Now that macOS is gone, can the contents of each target be factored outside to the top as one copy instead of three?

File: performance/SwiftUIPerformanceQuickstart/PerformanceExample/Podfile
Comment (paulb777) [inline performance/SwiftUIPerformanceQuickstart/PerformanceExample/Podfile:15]: Is 13.0 feasible?

File: performance/SwiftUIPerformanceQuickstart/PerformanceExample/Podfile
Comment (paulb777) [inline performance/SwiftUIPerformanceQuickstart/PerformanceExample/Podfile:14]: There should be Tests for tvOS as well

**Review by paulb777 — APPROVED** [review]
I'm ok to defer the comments except for the `pod deintegrate` to a future PR.

File: .github/workflows/performance.yml
Comment (granluo) [inline .github/workflows/performance.yml:57]: The test is for PRs with secrets access only. We might only run build for forks.

File: .github/workflows/performance.yml
Comment (granluo) [inline .github/workflows/performance.yml:69]: Could we use a [matrix](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstrategymatrix) here for ios and tvos

File: performance/SwiftUIPerformanceQuickstart/PerformanceExample/Shared/Process.swift
Comment (morganchen12) [reply performance/SwiftUIPerformanceQuickstart/PerformanceExample/Shared/Process.swift:21]: SGTM

File: .github/workflows/performance.yml
Comment (jrcrespoh) [reply .github/workflows/performance.yml:69]: I'm not sure how to incorporate it since `scheme`, `sdk`, and `destination` only correspond with either iOS or tvOS and shouldn't be mixed and matched...

File: .github/workflows/performance.yml
Comment (jrcrespoh) [reply .github/workflows/performance.yml:57]: Should I revert to checking for secrets?

Comment (granluo) [reply .github/workflows/performance.yml:57]: As discussed in the meeting, we can revert it to the previous if-else version, but `set output` to let the `have_secrets` be public to other steps.

File: .github/workflows/performance.yml
Comment (granluo) [reply .github/workflows/performance.yml:69]: Yep, I think to apply `matrix`, we might need a script to handle different params/flags. We can leave it like that before we have a script for all tests.

File: performance/SwiftUIPerformanceQuickstart/PerformanceExample/PerformanceExample.xcodeproj/project.pbxproj
Comment (jrcrespoh) [reply performance/SwiftUIPerformanceQuickstart/PerformanceExample/PerformanceExample.xcodeproj/project.pbxproj:24]: Done ✅

File: performance/SwiftUIPerformanceQuickstart/PerformanceExample/Podfile
Comment (jrcrespoh) [reply performance/SwiftUIPerformanceQuickstart/PerformanceExample/Podfile:4]: Done ✅

File: performance/SwiftUIPerformanceQuickstart/PerformanceExample/Podfile
Comment (jrcrespoh) [reply performance/SwiftUIPerformanceQuickstart/PerformanceExample/Podfile:15]: Some SwiftUI features are not available and would break the app.

File: performance/SwiftUIPerformanceQuickstart/PerformanceExample/Podfile
Comment (jrcrespoh) [reply performance/SwiftUIPerformanceQuickstart/PerformanceExample/Podfile:14]: Done ✅

File: .github/workflows/performance.yml
Comment (jrcrespoh) [reply .github/workflows/performance.yml]: Because these scripts will later be consolidated, I instead used `check_secrets.sh` to achieve this.

File: performance/SwiftUIPerformanceQuickstart/PerformanceExample/PerformanceExample.xcodeproj/project.pbxproj
Comment (paulb777) [inline performance/SwiftUIPerformanceQuickstart/PerformanceExample/PerformanceExample.xcodeproj/project.pbxproj:97]: Remove

File: performance/SwiftUIPerformanceQuickstart/PerformanceExample/PerformanceExample.xcodeproj/project.pbxproj
Comment (paulb777) [inline performance/SwiftUIPerformanceQuickstart/PerformanceExample/PerformanceExample.xcodeproj/project.pbxproj:139]: Remove.

**Review by paulb777 — APPROVED** [review]
The lingering `Pods` references are the result of running `pod deintegrate` with the project open in Xcode.

It can be fixed by editing manually or by opening the xcodeproj and deleting the `Pods` in the Project Navigator. It should be red.

File: performance/SwiftUIPerformanceQuickstart/PerformanceExample/Shared/Process.swift
Comment (paulb777) [inline performance/SwiftUIPerformanceQuickstart/PerformanceExample/Shared/Process.swift:187]: Consider adding an `#else`  with a fatal_error.

File: performance/SwiftUIPerformanceQuickstart/PerformanceExample/PerformanceExample.xcodeproj/project.pbxproj
Comment (jrcrespoh) [reply performance/SwiftUIPerformanceQuickstart/PerformanceExample/PerformanceExample.xcodeproj/project.pbxproj:97]: Done ✅

File: performance/SwiftUIPerformanceQuickstart/PerformanceExample/PerformanceExample.xcodeproj/project.pbxproj
Comment (jrcrespoh) [reply performance/SwiftUIPerformanceQuickstart/PerformanceExample/PerformanceExample.xcodeproj/project.pbxproj:139]: Done ✅

File: performance/SwiftUIPerformanceQuickstart/PerformanceExample/Shared/Process.swift
Comment (jrcrespoh) [reply performance/SwiftUIPerformanceQuickstart/PerformanceExample/Shared/Process.swift:187]: Done ✅

**Review by paulb777 — APPROVED** [review]
💯

**Pull request merged**
